scheduler/src/cook/kubernetes/api.clj

```diff
@@ -14,32 +14,59 @@
     (java.util.concurrent Executors ExecutorService)))
 
 
+(def cook-pod-label "twosigma.com/cook-scheduler-job")
+
 (def ^ExecutorService kubernetes-executor (Executors/newFixedThreadPool 2))
 
+(defn is-cook-scheduler-pod
+  "Is this a cook pod? Uses some-> so is null-safe."
+  [^V1Pod pod]
+  (some-> pod .getMetadata .getLabels (.get cook-pod-label)))
+
+(defn make-atom-updater
+  "Given a state atom, returns a callback that updates that state-atom when called with a key, prev item, and item."
+  [state-atom]
+  (fn
+    [key prev-item item]
+    (cond
+      (and (nil? prev-item) (not (nil? item))) (swap! state-atom (fn [m] (assoc m key item)))
+      (and (not (nil? prev-item)) (not (nil? item))) (swap! state-atom (fn [m] (assoc m key item)))
+      (and (not (nil? prev-item)) (nil? item)) (swap! state-atom (fn [m] (dissoc m key))))))
```

Comment:
Does this handle `"DELETED"` the right way? I thought the `object` on the watch was still non-nil in that case.

Reply:
No, it doesn't. (the bug is elsewhere, when we're encoding prev-item and item)

Reply:
Fixed now.